### PR TITLE
add into_vec

### PR DESCRIPTION
**Pull request by dev-ardi**
There's still some work to do but this would be the general idea

### CODE REVIEW

**Comment by algesten** [comment]
The idea is to use `into_reader()` for this.

```rs
    let mut bytes: Vec<u8> = Vec::with_capacity(len);
    resp.into_reader()
        .read_to_end(&mut bytes)?;
```

If that's unclear, we could look into making the doc better.

**Comment by dev-ardi** [comment]
Isn't this the same code that I sent?

File: src/response.rs [hunk 1]
```diff
@@ -481,6 +481,17 @@ impl Response {
         }
     }
 
+    pub fn into_vec(self) -> Result<Vec<u8>, Error> {
+        let size = self
+            .header("Content-Type")
```
Comment (johan-bjareholt) [inline src/response.rs:486]: Is this supposed to be "Content-Length"?
Also, what happens if e.g. "Transfer-Encoding: chunked" is used?
Using into_reader seems more convenient.

Comment (dev-ardi) [reply src/response.rs:486]: > Is this supposed to be "Content-Length"?

Oops it is :)

> Also, what happens if e.g. "Transfer-Encoding: chunked" is used?

Well in that case more allocations will be made, this is just a small optimization

**Comment by algesten** [comment]
Closing since we are moving to 3.x

**Pull request closed without merging**